### PR TITLE
Feature/models implementation

### PR DESCRIPTION
**Pull request by haider-sheikh**
Project Related all Models and association added. CanCan Ability model is also defined

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# This file is auto-generated from the current state of the database. Instead
```
Comment (hafsabatool7) [inline db/schema.rb]: Add schema to gitignore

Comment (haider-sheikh) [reply db/schema.rb]: file added to git ignore

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
```
Comment (hafsabatool7) [inline app/models/ability.rb]: Please remove these default comments.

File: app/models/team_user.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class TeamUser < ApplicationRecord
+  #attr_accessor :team
```
Comment (hafsabatool7) [inline app/models/team_user.rb]: Remove if not required

File: app/models/ability.rb (hunk 2 above)
Comment (haider-sheikh) [reply app/models/ability.rb]: removed..

File: app/models/team_user.rb (hunk 3 above)
Comment (haider-sheikh) [reply app/models/team_user.rb]: removed.

File: app/models/project.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Project < ApplicationRecord
+  has_many :project_teams
+  has_many :teams, through: :project_teams
+
+  has_many :project_users
```
Comment (hafsabatool7) [inline app/models/project.rb:8]: Do we not need association to users here?

File: app/models/project_user.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class ProjectUser < ApplicationRecord
+  belongs_to :project
```
Comment (hafsabatool7) [inline app/models/project_user.rb:2]: association to users?

File: app/models/team.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Team < ApplicationRecord
+  has_many :project_teams
+  has_many :projects, through: :project_teams
+
+  has_many :team_users
```
Comment (hafsabatool7) [inline app/models/team.rb:8]: Association for users

File: app/models/project.rb (hunk 4 above)
Comment (haider-sheikh) [reply app/models/project.rb:8]: user model is not created yet... so thats why i did not add it here

File: app/models/team.rb (hunk 6 above)
Comment (haider-sheikh) [reply app/models/team.rb:8]: user model is not created yet... so thats why i did not add it here

File: app/models/project_user.rb (hunk 5 above)
Comment (haider-sheikh) [reply app/models/project_user.rb:2]: user model is not created yet... so thats why i did not add it here

File: Gemfile [hunk 7]
```diff
@@ -17,6 +17,9 @@ gem 'webpacker', '~> 4.0'
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.7'
+
+#for authorization
+gem 'cancancan'
```
Comment (hafsabatool7) [inline Gemfile]: Add version

Comment (haider-sheikh) [reply Gemfile]: version added

File: Gemfile [hunk 8]
```diff
@@ -19,7 +19,7 @@ gem 'turbolinks', '~> 5'
 gem 'jbuilder', '~> 2.7'
 
 #for authorization
-gem 'cancancan'
+gem 'cancancan', '~> 3.1'
```
Comment (hafsabatool7) [inline Gemfile]: Use = instead of ~>

Comment (haider-sheikh) [reply Gemfile]: Fixed Version added for cancancan.

File: app/models/company.rb [hunk 9]
```diff
@@ -1,3 +1,9 @@
 class Company < ApplicationRecord
   has_many :users
+  has_many :projects
+  has_many :project_users
```
Comment (hafsabatool7) [inline app/models/company.rb]: Do we need this association?

**Pull request merged**